File: cos/cos.olympus/src/main/java/cos/olympus/game/SpellStrategy.java
package cos.olympus.game;

import cos.olympus.game.events.Damage;
import cos.olympus.game.events.Spell;
import cos.ops.OutOp;

import java.util.Collection;

public interface SpellStrategy {
    //        val action: SpellAction
    int id();
//        fun inZone(creature: Creature): Boolean
//        fun handle(time: Tsm, actions: ActionConsumer, map: GameMap): Boolean


    boolean onTick(int tick, Collection<OutOp> consumer, Collection<Damage> damages);

    boolean inZone(Creature cr);
    boolean isFinish();
    Spell spell();
}

File: cos/src/main/java/cos/olympus/game/strategy/SpellStrategy.kt
package cos.olympus.game.strategy

import cos.olympus.game.Creature
import cos.olympus.game.Damages
import cos.olympus.game.events.Spell

interface SpellStrategy {
    //        val action: SpellAction
    val id: Int


    //        fun inZone(creature: Creature): Boolean
    //        fun handle(time: Tsm, actions: ActionConsumer, map: GameMap): Boolean
    fun onTick(tick: Int, damages: Damages): Boolean

    fun inZone(cr: Creature): Boolean

    val finished: Boolean

    val spell: Spell
}
